### ISSUE TITLE
Support `depth` for `getKeys()`

### ISSUE THREAD

**Issue by melasculla** [unjs/unstorage#539]
### Describe the feature

hello there, im using unStorage to store images and files in filesystem im making it dynamic so can store files in different folders, but when i tried to get keys inside folder that have sub folders i got all items in sub folders too, i guess its expected behavior but would be great to have option to disable this behavior and get only current folder items

e.g.

```js
// incudeSubFolders: true
[
    "current.webp",
    "current.webp",
    "test:4.webp",
    "test:5.webp",
    "hello:world:1.webp",
    "hello:world:2.webp",
    "hello:world:3.webp"
]
```

```js
// incudeSubFolders: false
[
    "current.webp",
    "current.webp"
]
```

does that make sense? cause for now i need to filter keys and exclude all that have `:`
Thanks!

### Additional information

- [ ] Would you be willing to help implement this feature?

**Comment by pi0** [comment]
Nice suggestion. I have updated title to make it little more generic we can support `getKeys(base, { depth: 1 })` for same effect.

**Comment by melasculla** [comment]
Thank you so much! cant wait for it

**Comment by 43081j** [comment]
This can probably be closed now by #547

**Comment by pi0** [comment]
Thanks for help dear @43081j 

If you liked to help on next steps, I am thinking we start from most important one (fs/fs-lite) to also natively apply depth. Core filter will be duplicate (and we can later improve it by implementing `capabilities: {}`)
